src/ts/utils/event-hub.ts

```diff
@@ -100,28 +124,34 @@ export class EventHub extends EventEmitter {
     }
 
     /**
-     * Propagate events to children
+     * Propagate events to the whole subtree under this event hub.
+     * @internal
      */
-    private propagateToChildren(eventName: string, args: unknown[]) {
-
+    private propagateToThisAndSubtree(eventName: string, args: unknown[]) {
+        this.emitUnknown(eventName, args);
```

Comment:
Hi @martin31821

I was also struggling with `eventHub` in the new version and found your PR during my frantic googling. I also really needed a more dynamic way of communicating states to popup windows; thank you very much for implementing this! :)

I'm currently testing my project with this branch of yours and realized that my event listeners for `'userBroadcast'` events seem to receive the arguments wrapped in an array. I'm suspecting this line should actually read `this.emitUnknown(eventName, ...args);` (using a spread operator before `args`). Is that correct?

I'm not sure what this would mean for other (unknown) events, as the code without the spread operator seems to already have been present before your changes. But [emitUnknown seems to be defined using a spread operator for `args`](https://github.com/golden-layout/golden-layout/blob/32cb565516f4c465aaf628d4515b1c2636bcaf44/src/ts/utils/event-emitter.ts#L39).

Best Regards

Reply:
My bad. I did not include the rest operator in my comment in issue #647.

Reply:
Updated, good catch.

Reply:
Cool - that reflects the v1 behaviour